ReactNativeClient/lib/models/Tag.js

```diff
@@ -36,18 +42,48 @@ class Tag extends BaseItem {
 		);
 	}
 
+	static async childrenTagIds(parentId) {
+		const rows = await this.db().selectAll('SELECT id FROM tags WHERE parent_id = ?', [parentId]);
+		return rows.map(r => r.id);
+	}
+
 	// Untag all the notes and delete tag
-	static async untagAll(tagId) {
+	static async untagAll(tagId, options = null) {
+		if (!options) options = {};
+		if (!('deleteChildren' in options)) options.deleteChildren = true;
+
+		const tag = await Tag.load(tagId);
+		if (!tag) return; // noop
+
+		if (options.deleteChildren) {
+			const childrenTagIds = await Tag.childrenTagIds(tagId);
+			for (let i = 0; i < childrenTagIds.length; i++) {
+				await Tag.untagAll(childrenTagIds[i]);
+			}
+		}
+
 		const noteTags = await NoteTag.modelSelectAll('SELECT id FROM note_tags WHERE tag_id = ?', [tagId]);
 		for (let i = 0; i < noteTags.length; i++) {
 			await NoteTag.delete(noteTags[i].id);
 		}
 
-		await Tag.delete(tagId);
+		// We're recursing over all children already, so this call doesn't have to be recursive.
+		await Tag.delete(tagId, { deleteChildren: false });
```

Comment:
The purpose of this `deleteChildren` option is just to save a call to `Tag.childrenTagIds`? In that case, I think it's ok to remove it, as we probably won't get much better performance from it.

Reply:
Yes, that's correct.